messaging/Services/ConvertToIJEBackgroundWork.cs

```diff
@@ -154,130 +168,30 @@ private void HandleUpdateMessage(DeathRecordUpdateMessage message, IncomingMessa
-            CreateBFDRAckMessage<FetalDeathRecordAcknowledgementMessage>(message, databaseMessage, "FET", m => new FetalDeathRecordAcknowledgementMessage(m));
-            bool duplicateMessage = IncomingMessageLogItemExists(message.MessageId);
-            IncomingMessageLog previousMessage = LatestMessageByNCHSId(message.NCHSIdentifier);
-            if(!duplicateMessage) {
-                // Only log messages that are not duplicates
-                LogCommonMessage(message, databaseMessage);
-                // Only save if this is not a message with a duplicate ID and the previousMessage either does not exist or
-                // has an older timestamp than the message we are currently dealing with
-                if(previousMessage == null || message.MessageTimestamp > previousMessage.MessageTimestamp) {
-                    this._context.IJEItems.Add(ijeItem);
-                    this._context.SaveChanges();
-                }
-            }
-        }
-
-        private void LogCommonMessage(CommonMessage message, IncomingMessageItem databaseMessage) {
-            IncomingMessageLog entry = new IncomingMessageLog();
-            entry.MessageTimestamp = message.MessageTimestamp;
-            entry.MessageId = message.MessageId;
-            entry.JurisdictionId = databaseMessage.JurisdictionId;
-            //entry.NCHSIdentifier = message.NCHSIdentifier;
-            // TODO NCHS identifier isn't defined in CommonMessage, does it make sense to add it so we can use it here?
-            entry.NCHSIdentifier = "placeholder";
-            entry.StateAuxiliaryIdentifier = message.StateAuxiliaryId;
-            this._context.IncomingMessageLogs.Add(entry);
-            this._context.SaveChanges();
-        }
-
-        // TODO: once we move to VRDR STU3 and the common library version of VRDR, we can deprecate this function
-        private void LogMessage(DeathRecordSubmissionMessage message, IncomingMessageItem databaseMessage) {
-            IncomingMessageLog entry = new IncomingMessageLog();
-            entry.MessageTimestamp = message.MessageTimestamp;
-            entry.MessageId = message.MessageId;
-            entry.JurisdictionId = databaseMessage.JurisdictionId;
-            //entry.NCHSIdentifier = message.NCHSIdentifier;
-            // TODO NCHS identifier isn't defined in CommonMessage, does it make sense to add it so we can use it here?
-            entry.NCHSIdentifier = "placeholder";
-            entry.StateAuxiliaryIdentifier = message.StateAuxiliaryId;
-            this._context.IncomingMessageLogs.Add(entry);
-            this._context.SaveChanges();
-        }
-
-        private void CreateDeathAckMessage(BaseMessage message, IncomingMessageItem databaseMessage) {
+        private static void CreateAckMessage<T>(CommonMessage message, IncomingMessageItem databaseMessage, string eventType, Func<CommonMessage, T> createMessage, ApplicationDbContext _context) where T: CommonMessage
```

Comment:
My initial reaction to this is that it would create ack messages that are too generic. We used the specific Ack types so that it would be clear if an acknowledgement is for a birth message, fetal death message, or death message. Otherwise, once the jurisdiction has the fhir message in their system it would be unclear what the message type is. Does the MessageType get set correctly? or would it default to a default message type?